### PR TITLE
add all the flags

### PR DESCRIPTION
**Pull request by czarcas7ic**
Kept the -m and -t flags that automate the setup.

Gave every user input a corresponding flag with default values. If the flag is called but nothing is stated after the flag, the default is used. If the flag is not used, user input will be required at the corresponding step.

Screen Shot 2022-04-19 at 8 57 52 PM

### CODE REVIEW

**Comment by czarcas7ic** [comment]
Changed the delimiter from a comma to a semicolon in order to allow multiple entries for ports which require commas in the config and app.toml

Made it so that if no flags are specified, the script is interactive (no defaults). If one or more is specified, all defaults are applied unless otherwise specified by a user inputted flag

File: osmosis-installer.py [hunk 1]
```diff
@@ -5,17 +5,63 @@
...
 from os import remove
-from sys import argv
 
-remove(argv[0])
-
-parser = argparse.ArgumentParser(description="Use default settings")
-parser.add_argument('-m', action='store_true', help='use default settings with no input for mainnet')
-parser.add_argument('-t', action='store_true', help='use default settings with no input for mainnet')
 
+#remove(argv[0])
```
Comment (p0mvn) [inline osmosis-installer.py:12]: what is this comment?

File: osmosis-installer.py [hunk 2]
```diff
@@ -5,17 +5,63 @@
...
+            return text[2:].splitlines()
+        # this is the RawTextHelpFormatter._split_lines
+        return argparse.HelpFormatter._split_lines(self, text, width)
+fmt = lambda prog: CustomHelpFormatter(prog,max_help_position=30)
+
+osmo_home = subprocess.run(["echo $HOME/.osmosisd"], capture_output=True, shell=True, text=True).stdout.strip()
+
+parser = argparse.ArgumentParser(description="Osmosis Installer",formatter_class=fmt)
+auto = parser.add_argument_group('Automated')
+auto.add_argument('-m', '--mainnet-default', action='store_true', help='R|Use all default settings with no input for mainnet\n ', dest="m")
```
Comment (p0mvn) [inline osmosis-installer.py:30]: q: what is "R" in `'R|Use all..` ?

File: osmosis-installer.py [hunk 3]
```diff
@@ -204,7 +254,13 @@ def startReplayNow():
 1) Yes, start cosmovisor as a background service and begin replay
 2) No, exit and start on my own (will still auto update at upgrade heights)
     """+ bcolors.ENDC)
-    startNow = input(bcolors.OKGREEN + 'Enter Choice: '+ bcolors.ENDC)
+    if args.sr == True :
```
Comment (p0mvn) [inline osmosis-installer.py:257]: Please choose more descriptive names instead of abbreviations

File: osmosis-installer.py [hunk 4]
```diff
@@ -161,8 +207,12 @@ def cosmovisorInit ():
 2) No, just set up an osmosisd background service (recommended)
 3) Don't install cosmovisor and don't set up a background service
     """+ bcolors.ENDC)
-    if args.m == True :
+    if args.cvs == "cosmoservice" :
```
Comment (p0mvn) [inline osmosis-installer.py:210]: more descriptive names for all similar arg variables please

**Review by p0mvn — COMMENTED** [review]
Skimmed over - LGTM, I would only suggest considering more descriptive names for the args, please. It's hard to read the abbreviations

File: osmosis-installer.py (hunk 1 above)
Comment (czarcas7ic) [reply osmosis-installer.py:12]: Thanks for catching, this should actually be uncommented. Its what "self-destructs" the python file. I comment it out while testing so it doesn't delete itself.

File: osmosis-installer.py (hunk 2 above)
Comment (czarcas7ic) [reply osmosis-installer.py:30]: This is from the CustomHelpFormatter which allows formatting (such as new lines) for the help menu (seen by using the -h flag). Without it the help menu is very hard to read.

File: osmosis-installer.py (hunk 3 above)
Comment (czarcas7ic) [reply osmosis-installer.py:257]: Great call, I have no idea why I didn't do this from the start. Made it super difficult to read but now its much better, thanks!

File: osmosis-installer.py [hunk 5]
```diff
@@ -5,17 +5,64 @@
...
+args = ['default', 'nothing', 'everything']; both.add_argument('-pr', '--prune', type = str, choices=args, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
+args = ['cosmoservice', 'osmoservice', 'noservice']; both.add_argument('-cvs', '--cosmovisor-service', type = str, choices=args, default='osmoservice', help='R|Start with cosmovisor systemctl service, osmosisd systemctl service, or exit without creating or starting a service \nDefault: "osmoservice" '+str(args), dest="cosmovisorService")
+testnet = parser.add_argument_group('Testnet only')
+#testnet
+args = ['snapshot', 'exit']; testnet.add_argument('-dst', '--data-sync-test', type = str, choices=args, default='snapshot', help='R|Data sync options \nDefault: "snapshot" '+str(args)+'\n ', dest="dataSyncTestnet")
+args = ['pruned', 'archive']; testnet.add_argument('-stt', '--snapshot-type-test', type = str, choices=args, default='pruned', help='R|Snapshot type \nDefault: "pruned" '+str(args)+'\n ', dest="snapshotTypeTestnet")
```
Comment (p0mvn) [inline osmosis-installer.py:47]: I'm not super familiar with Python formatting rules but I'm wondering if it is possible to format these somehow? E.g. split into separate lines for readability?

**Review by p0mvn — COMMENTED** [review]
New arg names look great! One more readability specific question

File: osmosis-installer.py (hunk 5 above)
Comment (czarcas7ic) [reply osmosis-installer.py:47]: Do you mean specifically separate the args from the add_argument lines?

Comment (p0mvn) [reply osmosis-installer.py:47]: Yes, and change the names, for example:
```
args = ['default', 'nothing', 'everything']; both.add_argument('-pr', '--prune', type = str, choices=args, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
```

```
pruningStrateegyArgChoices = [
     'default', 
    'nothing', 
    'everything'
]
both.add_argument('-pr', '--prune', type = str, choices=pruningStrateegyArgChoices, default='everything', help='R|Pruning settings \nDefault: "everything" '+str(args)+'\n ', dest="pruning")
```

Comment (nikever) [reply osmosis-installer.py:47]: An idiomatic way to improve readability in python could be:

```python
both.add_argument(
    '-s', 
    '--swap', 
    type = bool, 
    default=True, 
    help='R|Use swap if less than 32Gb RAM are detected \nDefault (bool): True\n ', 
    dest="swapOn")
```

**Comment by czarcas7ic** [comment]
@p0mvn Just fixed the flags, if you could give it a 2 minute glance before I merge that would be awesome

**Review by p0mvn — APPROVED** [review]
Awesome work!

**Pull request merged**